concordium-node/src/stats_export_service.rs

```diff
@@ -44,6 +46,27 @@ impl PrometheusStateData {
     }
 }
 
+/// Wrapper for implementing the prometheus::Collector trait for
+/// InFlightRequestsCounter, which syncs the prometheus gauge with the counter
+/// on scrape.
+struct GrpcInFlightRequestsCollector {
+    /// The prometheus gauge exposed.
+    pub gauge:   IntGauge,
+    /// Counter used to track in flight requests by
+    /// `tower_http::metrics::InFlightRequestsLayer`.
+    pub counter: InFlightRequestsCounter,
+}
```

Comment:
Why is the struct private but the fields public?
Also, what are the semantics of that? 🤔

Reply:
The semantics for anything like this is that all components of the path need to be visible for you to access the item.

But for this case there is no difference between pub and non/pub fields I think. Since the type cannot be exported we cannot access its fields outside of the module anyhow.

Reply:
Right, although it makes no difference, ill remove these to make the definition less noisy